draft-irtf-cfrg-frost.md

```diff
@@ -1672,3 +1672,28 @@ S3 sig_share: 9651d355ca1dea2557ba1f73e38a9f4ff1f1afc565323ef27f88a9d
 sig: 02dfba781e17b830229ae4ed22ebe402873683d9dfd945d01762217fb3172c2a
 71f83a8d1a3efd188c04d41cf48a716e11b8eff38607023c1f9bb0d36fe1d9f2e9
 ~~~
+
+
+# Random Scalar Generation {#random-scalar}
+
+Two popular algorithms for generating a random integer uniformly distributed in
+the range \[0, G.Order() -1\] are as follows:
+
+## Rejection Sampling
+
+Generate a random byte array with `Ns` bytes, and attempt to map to a Scalar
+by calling `DeserializeScalar`. If it works, return the result. If it fails,
+try again with another random byte array, until the procedure succeeds.
+
+Note the that the Scalar size might be some bits smaller than the array size,
+which can result in the loop iterating more times than required. In that case
+it's acceptable to set the high-order bits to 0 before calling `DeserializeScalar`,
+but care must be taken to not set to zero more bits than required. For example,
+in the `FROST(Ed25519, SHA-512)` ciphersuite, the order has 253 bits while
+the array has 256; thus the top 3 bits of the last byte can be set to zero.
+
+## Wide Reduction
+
+Generate a random byte array with `Ns * 2` bytes, and interpret it as a
```

Comment:
Is this wide enough to remove bias? The [hash-to-curve draft](https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-hash-to-curve-16#section-5) uses the following expression to determine the width:

```
L = ceil((ceil(log2(p)) + k) / 8)
```

where p is the prime associated with the field and k is the target security level. This yields a bias of at most 2^-k. I would prefer if we wrote the width in a similar way here.

Reply:
Thanks for the reference. It turns out what I wrote was even more than required.
Since the RFC does not mention security levels, I think it would be confusing to use that expression as is. So I thought of using the hash-to-curve [suggestion](https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-hash-to-curve-16#section-8.9) of setting `k = ceil(log2(r) / 2)` and substituting in the expression (it ends up being basically `3/2 * Ns`). This was done in 7ae7e944df7, let me know if it makes sense.